pkg/scheduler/schedule_one.go

```diff
@@ -283,6 +283,15 @@ func (sched *Scheduler) bindingCycle(
 		return status
 	}
 
+	// All other extension points after here aren't supposed to fail in a usual state.
+	// i.e., They could return error only when some unexpected temporal errors happen. (e.g., kube-apiserver failure)
+	// So, if Pods are failed at one of them and going to the scheduling queue, the scheduling queue always pushes them to activeQ/backoffQ, instead of the unschedulable pod pool.
+	// This is because such Pods won't be schedulable by any certain cluster events, and thus scheduling hints aren't useful to determine when to push those Pods to activeQ/backoffQ.
+	//
+	// We can call Done() here because Pods failed at all other extension points after here won't use the scheduling hints as written.
```

Comment:
```suggestion
	// We can call Done() here because Pods failed at all other extension points after here will be retried regardless of the events that happened in the cluster.
```